graveler/graveler.go

```diff
@@ -620,42 +621,46 @@ func (g *Graveler) CreateBranch(ctx context.Context, repositoryID RepositoryID,
 
 func (g *Graveler) UpdateBranch(ctx context.Context, repositoryID RepositoryID, branchID BranchID, ref Ref) (*Branch, error) {
 	res, err := g.branchLocker.MetadataUpdater(ctx, repositoryID, branchID, func() (interface{}, error) {
-		reference, err := g.RefManager.RevParse(ctx, repositoryID, ref)
-		if err != nil {
-			return nil, err
-		}
-
-		curBranch, err := g.RefManager.GetBranch(ctx, repositoryID, branchID)
-		if err != nil {
-			return nil, err
-		}
-		// validate no conflict
-		// TODO(Guys) return error only on conflicts, currently returns error for any changes on staging
-		iter, err := g.StagingManager.List(ctx, curBranch.StagingToken)
-		if err != nil {
-			return nil, err
-		}
-		defer iter.Close()
-		if iter.Next() {
-			return nil, ErrConflictFound
-		}
-
-		newBranch := Branch{
-			CommitID:     reference.CommitID(),
-			StagingToken: curBranch.StagingToken,
-		}
-		err = g.RefManager.SetBranch(ctx, repositoryID, branchID, newBranch)
-		if err != nil {
-			return nil, err
-		}
-		return &newBranch, nil
+		return g.updateBranch(ctx, repositoryID, branchID, ref)
 	})
 	if err != nil {
 		return nil, err
 	}
 	return res.(*Branch), nil
 }
 
+func (g *Graveler) updateBranch(ctx context.Context, repositoryID RepositoryID, branchID BranchID, ref Ref) (*Branch, error) {
```

Comment:
I took the same `UpdateBranch` as is and moved it to `updateBranch` without a lock so that `CommitExistingMetaRange` can use it. The "validate no conflict here is the check that staging is empty. @nopcoder Do we wan't that to be optional?

Reply:
Don't think there is currently a flow that doesn't need this validation. The strange part is that we have a function that should update the branch information and it also validates you don't have anything in the stage area.

Reply:
I think it should be part of gravelers update branch functionality.
Otherwise, what would you suggest?